src/df.jl

```diff
@@ -0,0 +1,37 @@
+"""
+    `@df d x`
+
+Convert every symbol in the expression `x` with the respective column in `d` if it exists.
+
+If you want to avoid replacing the symbol, escape it with `^`.
+
+`NA` values are replaced with `NaN` for columns of `Float64` and `""` or `Symbol()`
+for strings and symbols respectively.
+"""
+macro df(d, x)
+    esc(_df(d,x))
+end
+
+_df(d, x) = x
+
+function _df(d, x::Expr)
```

Comment:
Perhaps here we could have some convenience functions here that are useful for plotting with DataFrames. Like allowing the symbol `:nrow` to be replaced by `size(x,1)`. 
Would it also be nice to be able to specify columns by number, like `cols(1:4)` and get `reshape(names(d)[x],1,:)`?

Reply:
That's an interesting idea, especially for `cols(1:4)` (for let's say a `corrplot`). Two questions:
- what's an example use of `size(x,1)` in a plot call?
- should we go for something like: `df` gets replaced with the `DataFrame`? In that case maybe `:nrow` can be replaced by `nrow(df)` (which would work for free) and we'd get access to all the `DataFrame` functions (though, to be honest, I'm not sure how many would actually be useful)

Reply:
- `marker_z = 1:size(x,1)`
- No, I'm not sure that would be necessary or clean

Reply:
Agreed on point 2. About point 1, there's the issue that `marker_z = 1:(:nrow)` is not the prettiest syntax, plus we create even more ambiguity with the symbols: what if a column is called `:nrow`? Maybe simply `nrow` and `ncol` are used (normal function names act on the DataFrame and symbols are for columns). Now `cols(1:ncol)` does the full `corrplot`.

Reply:
Are these extra features compatible to the plan to support a variety of data table tapes?
If not: Do we even still pursue this plan?

Reply:
My take - we must pursue that long-term, but it seems early to me to start second-guessing what the interface will be.

Reply:
@piever you have a point on nrow and friends, let's keep that out of it. When you say "now cols(1:ncol) does" do you mean that you've already added the functionality but not pushed it?

Reply:
`cols` yes, `nrow` it depends (or at least it would require some thinking, depending on which plan we choose). I think the plan to support multiple data source has at least two possible implementations:
- `@df` converts the first argument to a DataFrame (using IterableTables) and then proceeds as usual: anything that'd work on a DataFrame keeps working. It could be inefficient to convert the whole thing though (plotting a big DataFrame is rare, but I often use `groupapply` on pretty large data)
- the only thing that is DataFrame specific, gets done with IterableTables: `df[col]` is changed to `[getindex(s, col) for s in df]` (I need to see how it'd work with missing data)

Getting plan 2 to work with `groupapply` may require some thinking, so I thought it's better to postpone this decision, until adjusting `groupapply` to the new syntax.

Reply:
I like 2.

Reply:
> When you say "now cols(1:ncol) does

It's more like wishful thinking and poor phrasing... I meant with the hypothetical implementations. Still, if we decide that `nrow` and `ncol` are left out for the time being, implementing `cols` is pretty easy, I should be able to add it and push today.